scalding-beam/src/main/scala/com/twitter/scalding/beam_backend/BeamOp.scala

```diff
@@ -51,9 +78,131 @@ object BeamOp extends Serializable {
+            input
+              .apply(TupleToKV[K, V](kryoCoder, kryoCoder))
+              .apply(ParDo.of(MapSideAggregator[K, V](size, semigroup)))
+              .apply(KVToTuple[K, V])
+          }
+        },
+        kryoCoder)
+    }
+  }
+
+  case class SortGroupedValues[K, V](implicit ordV: Ordering[V], kryoCoder: KryoCoder) extends PTransform[PCollection[KV[K, java.lang.Iterable[V]]], PCollection[KV[K, java.lang.Iterable[V]]]] {
+    override def expand(input: PCollection[KV[K, lang.Iterable[V]]]): PCollection[KV[K, lang.Iterable[V]]] = {
+      input
+        .apply(ParDo.of(MapFn[KV[K, java.lang.Iterable[V]], KV[K, java.lang.Iterable[KV[WrappedComparable[V], Unit]]]]{ elem =>
+          KV.of(elem.getKey, elem.getValue.asScala.map(x => KV.of(WrappedComparable(x, ordV), ())).asJava)
+        })).setCoder(KvCoder.of(kryoCoder, IterableCoder.of(KvCoder.of(WrappedAtomicCoder[V](ordV, OrderedSerializationCoder(ordV)), kryoCoder))))
+        .apply(SortValues.create[K, WrappedComparable[V], Unit](BufferedExternalSorter.options()))
+        .apply(ParDo.of(MapFn[KV[K, java.lang.Iterable[KV[WrappedComparable[V], Unit]]], KV[K, java.lang.Iterable[V]]]{ elem =>
+          KV.of(elem.getKey, elem.getValue.asScala.map(_.getKey.value).asJava)
+        })).setCoder(KvCoder.of(kryoCoder, IterableCoder.of(OrderedSerializationCoder(ordV))))
+    }
+  }
+
+  case class TupleToKV[K, V](kCoder: Coder[K], vCoder: Coder[V]) extends PTransform[PCollection[(K, V)], PCollection[KV[K, V]]] {
+    override def expand(input: PCollection[(K, V)]): PCollection[KV[K, V]] = {
+      input
+        .apply(MapElements.via[(K, V), KV[K, V]](new SimpleFunction[(K, V), KV[K, V]]() {
+          override def apply(input: (K, V)): KV[K, V] = KV.of(input._1, input._2)
+        })).setCoder(KvCoder.of(kCoder, vCoder))
+    }
+  }
+
+  case class KVToTuple[K, V](implicit kryoCoder: KryoCoder) extends PTransform[PCollection[KV[K, V]], PCollection[(K, V)]] {
+    override def expand(input: PCollection[KV[K, V]]): PCollection[(K, V)] = {
+      input
+        .apply(MapElements.via[KV[K, V], (K, V)](new SimpleFunction[KV[K, V], (K, V)]() {
+          override def apply(input: KV[K, V]): (K, V) = (input.getKey, input.getValue)
+        })).setCoder(kryoCoder)
+    }
+  }
+
+  case class WrappedComparable[A](value: A, ord: Ordering[A]) extends Comparable[WrappedComparable[A]] {
+    override def compareTo(o: WrappedComparable[A]): Int = ord.compare(value, o.value)
+    override def hashCode(): Int = value.hashCode()
+    override def equals(obj: Any): Boolean =
+      obj match {
+        case WrappedComparable(objValue, _) => value.equals(objValue)
+        case _ => false
+      }
+  }
+
+  case class WrappedAtomicCoder[A](ord: Ordering[A], prev: AtomicCoder[A]) extends AtomicCoder[WrappedComparable[A]] {
+    override def encode(
+      value: WrappedComparable[A],
+      outStream: OutputStream): Unit = prev.encode(value.value, outStream)
```

Comment:
Add `assert` on `ord` being same as wrapped ordering

Reply:
I think we can remove the whole class since we need a different strategy to do secondary sorts anyway...